pkg/monitoring/virt-controller/metrics/vm_metrics.go

```diff
@@ -0,0 +1,56 @@
+package metrics
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/prometheus/client_golang/prometheus"
+	ioprometheusclient "github.com/prometheus/client_model/go"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+var (
+	vmMetrics = []operatormetrics.Metric{
+		vmsCreatedCounter,
+	}
+
+	vmsCreatedCounter = operatormetrics.NewCounterVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vm_created_total",
+			Help: "Amount of VMs created, broken down by namespace.",
+		},
+		[]string{"namespace"},
+	)
+
+	vmMap = map[string]bool{}
```

Comment:
So we're essentially keeping state in the virt-controller here, which (correct me if I'm wrong) means that the created VM metric will start lying as soon as the virt-controller pod dies?

Reply:
this state is to prevent "concurrent" requests when the VM is first created
because the `vm.Status.Conditions` are not set yet in these (which would make this function being skipped as I mentioned in https://github.com/kubevirt/kubevirt/pull/10418#discussion_r1364694364)

Reply:
I mean, on a system with existing VMs, when the virt-controller dies because of maintenance or something,
the metric would reset to 0 (lie). Is that not the case?

Reply:
That is expected with counters, Prometheus handles those counter resets

Reply:
So we would be okay with this metric not reported the correct amount of VMs after a restart?

Reply:
I wouldn't put it in that way. We want the metric correctly report the number of VMs created by that instance of 'virt-controller'